src/neo/Trie/MPT/MPTNode.cs

```diff
@@ -0,0 +1,103 @@
+using System.IO;
+using System.Text;
+using Neo.Cryptography;
+using Neo.IO;
+
+namespace Neo.Trie.MPT
+{
+    public enum NodeType
+    {
+        FullNode,
+        ShortNode,
+        HashNode,
+        ValueNode,
+    }
+
+    public class NodeFlag
+    {
+        public byte[] Hash;
+        public bool Dirty;
+
+        public NodeFlag()
+        {
+            Dirty = true;
+        }
+    }
+
+    public abstract class MPTNode : ISerializable
+    {
+        public NodeFlag Flag;
+        protected NodeType nType;
+
+        protected abstract byte[] calHash();
+
+        public virtual byte[] GetHash()
+        {
+            if (!Flag.Dirty) return Flag.Hash;
+            Flag.Hash = calHash();
+            Flag.Dirty = false;
+            return (byte[])Flag.Hash.Clone();
+        }
+
+        public void ResetFlag()
+        {
+            Flag = new NodeFlag();
+        }
+
+        public int Size { get; }
```

Comment:
I think we should use `public virtual int Size`, and overide the `Size` attribute in each child class.

Reply:
👌